launcher/ui/GuiUtil.cpp

```diff
@@ -75,10 +99,35 @@ std::optional<QString> GuiUtil::uploadPaste(const QString& name, const QString&
 
             if (response != QMessageBox::Yes)
                 return {};
+
+            if (baseUrl.toString() == "https://api.mclo.gs" && text.count("\n") > MaxMclogsLines) {
+                auto truncateResponse =
+                    CustomMessageBox::selectable(parentWidget, QObject::tr("Confirm Truncate"),
+                                                 QObject::tr("The log exceeds mclo.gs' limit: %1 lines (max %2).\n"
+                                                             "Prism can keep the first %3 and last %4 lines, trimming the middle.\n\n"
+                                                             "If you choose 'No', mclo.gs will only keep the first %2 lines, cutting off "
+                                                             "potentially useful info like crashes at the end.\n\n"
+                                                             "Proceed with Prism's truncation?")
+                                                     .arg(text.count("\n"))
+                                                     .arg(MaxMclogsLines)
+                                                     .arg(InitialMclogsLines)
+                                                     .arg(FinalMclogsLines),
+                                                 QMessageBox::Warning, QMessageBox::Yes | QMessageBox::No, QMessageBox::No)
+                        ->exec();
+
+                if (truncateResponse == QMessageBox::Yes)
+                    shouldTruncate = true;
+            }
         }
     }
 
-    std::unique_ptr<PasteUpload> paste(new PasteUpload(parentWidget, text, pasteCustomAPIBaseSetting, pasteTypeSetting));
+    QString textToUpload;
+    if (shouldTruncate)
+        textToUpload = truncateLogForMclogs(text);
+    else
+        textToUpload = text;
```

Comment:
```suggestion
    QString textToUpload = text;
    if (shouldTruncate) {
        textToUpload = truncateLogForMclogs(text);
    }
```

Reply:
I find that suggestion to be more awkward to read that the original code.
A ternary would be much more appropriate in that context.

PS: I was originally gonna comment about how this might just trigger 2 copies instead of one but somehow qt decided `operator=` would be reference counted instead of a copy behind the scene.